src/model.rs

```diff
@@ -2,129 +2,172 @@ use std::collections::BTreeMap;
 pub enum TraitType {
+    /// A spreadscaler trait
     #[serde(rename = "spreadscaler")]
-    Spreadscaler,
+    SpreadScaler,
+    /// A linkdef trait
     #[serde(rename = "linkdef")]
     Linkdef,
 }
 
+/// Properties for defining traits
 #[derive(Debug, Serialize, Deserialize)]
-pub struct LinkdefValueItems {
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub uri: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub address: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub port: Option<i32>,
-}
-
-fn default_weight() -> Option<i32> {
-    Some(DEFAULT_WEIGHT)
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Spread {
-    pub name: String,
-    pub requirements: BTreeMap<String, String>,
-    #[serde(default = "default_weight", skip_serializing_if = "Option::is_none")]
-    pub weight: Option<i32>,
+#[serde(untagged)]
+pub enum TraitProperty {
+    Linkdef(LinkdefProperty),
+    SpreadScaler(SpreadScalerProperty),
 }
 
+/// Properties for linkdefs
 #[derive(Debug, Serialize, Deserialize)]
 pub struct LinkdefProperty {
+    /// The target this linkdef applies to. This should be the name of an actor component
     pub target: String,
-    pub values: LinkdefValueItems,
+    /// Values to use for this linkdef
+    pub values: BTreeMap<String, String>,
 }
 
+/// Properties for spread scalers
 #[derive(Debug, Serialize, Deserialize)]
 pub struct SpreadScalerProperty {
-    pub replicas: i32,
+    /// Number of replicas to scale
+    pub replicas: usize,
+    /// Requirements for spreading throse replicas
     pub spread: Vec<Spread>,
 }
 
+/// Configuration for various spreading requirements
 #[derive(Debug, Serialize, Deserialize)]
-#[serde(untagged)]
-pub enum TraitPropertyItem {
-    LinkdefProperty(LinkdefProperty),
-    SpreadScalerProperty(SpreadScalerProperty),
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Traits {
-    #[serde(rename = "type")]
-    pub trait_type: TraitType,
-    pub properties: TraitPropertyItem,
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct ComponentItem {
+pub struct Spread {
+    /// The name of this spread requirement
     pub name: String,
-    #[serde(rename = "type")]
-    pub component_type: ComponentType,
-    pub properties: Properties,
+    /// An arbitrary map of labels to match on for scaling requirements
+    pub requirements: BTreeMap<String, String>,
+    /// An optional weight for this spread. Higher weights are given more precedence
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub traits: Option<Vec<Traits>>,
-}
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Components {
-    pub components: Vec<ComponentItem>,
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Manifest {
-    #[serde(rename = "apiVersion")]
-    pub api_version: String,
-    pub kind: String,
-    pub metadata: Metadata,
-    pub spec: Components,
+    pub weight: Option<usize>,
```

Comment:
nit: skip if none

Reply:
This is already there, the diff just makes it looks like it hasn't